torch/linalg/__init__.py

```diff
@@ -2251,3 +2251,42 @@
     >>> torch.dist(Q.mT @ Q, torch.eye(4))
     tensor(6.2158e-07)
 """)
+
+vecdot = _add_docstr(_linalg.linalg_vecdot, r"""
+linalg.vecdot(x, y, *, dim=-1, out=None) -> Tensor
+
+Computes the vector product of two batches of vectors along a dimension.
+
+In symbols, this function computes
+
+.. math::
+
+    \sum_{i=1}^n x_iy_i.
+
+over the dimension :attr:`dim`.
+
+Supports input of half, bfloat16, float, double, cfloat, cdouble and integral dtypes.
```

Comment:
If it supports everything do we need to mention it?

I think it's fine to be explicit here. Also, many other linalg functions have a note on supported dtypes.

Reply:
Leaving it as "better explicit than implicit". Also, it does support `bool`, but it's not announced here. If I had to choose I'd disallow integral and boolean types, but left them because why not.